snow/engine/snowman/bootstrap/bootstrapper.go

```diff
@@ -513,13 +496,25 @@ func (b *Bootstrapper) Ancestors(ctx context.Context, nodeID ids.NodeID, request
 			zap.Stringer("expectedBlkID", wantedBlkID),
 			zap.Stringer("blkID", actualID),
 		)
+		b.PeerTracker.RegisterFailure(nodeID)
 		return b.fetch(ctx, wantedBlkID)
 	}
 
-	blockSet := make(map[ids.ID]snowman.Block, len(blocks))
+	var (
+		numBytes = len(requestedBlock.Bytes())
+		blockSet = make(map[ids.ID]snowman.Block, len(blocks))
+	)
 	for _, block := range blocks[1:] {
+		numBytes += len(block.Bytes())
 		blockSet[block.ID()] = block
 	}
+
+	var (
+		requestLatency = time.Since(requestTime).Seconds() + epsilon
+		bandwidth      = float64(numBytes) / requestLatency
+	)
+	b.PeerTracker.RegisterResponse(nodeID, bandwidth)
```

Comment:
Should we try to calculate the "useful" bandwidth? Technically someone could provide the request block and then random ancestors and be considered good even though the ancestors were useless.

Reply:
added a todo